Build from source fails on system without python2 installed since 850dbda

Hello,

since commit 850dbda5c374defe3cdafb17941888b90362ab52 build from source is broken on newer systems without python2 installed.

This is probably because node-sass is depending on node-gyp v3.8.0 which doesn't support python3 (https://github.com/sass/node-sass/issues/3013).

It is possible to install python2 on most systems, but it's EOL and deprecated for over a year now!
So better solution would be to override the node-sass dependency with a newer version of node-gyp.

**What did you do?**

Build prometheus from source using `make build` on a system without python2 installed.

**What did you expect to see?**

A succesful  build.

**What did you see instead? Under which circumstances?**

Build fails with error -> see Logs

**Environment**

Build in Docker container based on Debian 11 without installing python2 package.

```Dockerfile
FROM debian:bullseye-slim
WORKDIR ${GOPATH}/src
RUN apt-get update && apt-get install -y golang git curl bash build-essential
RUN curl -fsSL https://deb.nodesource.com/setup_15.x | bash -
RUN apt-get update && apt-get install -y nodejs
RUN curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor | tee /usr/share/keyrings/yarnkey.gpg >/dev/null
RUN echo "deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main" | tee /etc/apt/sources.list.d/yarn.list
RUN apt-get -y install yarn
RUN git clone http://github.com/prometheus/prometheus.git --single-branch --branch main --depth=1
WORKDIR ${GOPATH}/src/prometheus
RUN make build
```


* Logs:

Providing only /usr/bin/python3 
```bash
[4/4] Building fresh packages...
error /src/prometheus/web/ui/react-app/node_modules/node-sass: Command failed.
Exit code: 1
Command: node scripts/build.js
Arguments: 
Directory: /src/prometheus/web/ui/react-app/node_modules/node-sass
Output:
Building: /usr/bin/node /src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   '/usr/bin/node',
gyp verb cli   '/src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library='
gyp verb cli ]
gyp info using node-gyp@3.8.0
gyp info using node@15.14.0 | linux | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21)
gyp verb `which` failed  python2 Error: not found: python2
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21) {
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` failed Error: not found: python
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21)
gyp verb `which` failed  python Error: not found: python
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21) {
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/src/prometheus/web/ui/react-app/node_modules/node-gyp/lib/configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (/src/prometheus/web/ui/react-app/node_modules/node-gyp/lib/configure.js:406:16)
gyp ERR! stack     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:16)
gyp ERR! stack     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp ERR! stack     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp ERR! stack     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqCallback.oncomplete (node:fs:193:21)
gyp ERR! System Linux 5.8.0-50-generic
gyp ERR! command "/usr/bin/node" "/src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /src/prometheus/web/ui/react-app/node_modules/node-sass
gyp ERR! node -v v15.14.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
Build failed with error code: 1
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
make: *** [Makefile:36: web/ui/react-app/node_modules] Error 1
```
Linking /usr/bin/python3 to /usr/bin/python

```bash
[4/4] Building fresh packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error /src/prometheus/web/ui/react-app/node_modules/node-sass: Command failed.
Exit code: 1
Command: node scripts/build.js
Arguments: 
Directory: /src/prometheus/web/ui/react-app/node_modules/node-sass
Output:
Building: /usr/bin/node /src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   '/usr/bin/node',
gyp verb cli   '/src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library='
gyp verb cli ]
gyp info using node-gyp@3.8.0
gyp info using node@15.14.0 | linux | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21)
gyp verb `which` failed  python2 Error: not found: python2
gyp verb `which` failed     at getNotFoundError (/src/prometheus/web/ui/react-app/node_modules/which/which.js:13:12)
gyp verb `which` failed     at F (/src/prometheus/web/ui/react-app/node_modules/which/which.js:68:19)
gyp verb `which` failed     at E (/src/prometheus/web/ui/react-app/node_modules/which/which.js:80:29)
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/which/which.js:89:16
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/index.js:42:5
gyp verb `which` failed     at /src/prometheus/web/ui/react-app/node_modules/isexe/mode.js:8:5
gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:193:21) {
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python /usr/bin/python
gyp ERR! configure error 
gyp ERR! stack Error: Command failed: /usr/bin/python -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                       ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack 
gyp ERR! stack     at ChildProcess.exithandler (node:child_process:326:12)
gyp ERR! stack     at ChildProcess.emit (node:events:369:20)
gyp ERR! stack     at maybeClose (node:internal/child_process:1067:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
gyp ERR! System Linux 5.8.0-50-generic
gyp ERR! command "/usr/bin/node" "/src/prometheus/web/ui/react-app/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /src/prometheus/web/ui/react-app/node_modules/node-sass
gyp ERR! node -v v15.14.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
Build failed with error code: 1
make: *** [Makefile:36: web/ui/react-app/node_modules] Error 1
The command '/bin/sh -c make build' returned a non-zero code: 2
```

--- a/web/ui/react-app/package.json
+++ b/web/ui/react-app/package.json
@@ -32,7 +32,7 @@
     "jsdom": "^16.4.0",
     "moment": "^2.24.0",
     "moment-timezone": "^0.5.23",
-    "node-sass": "4.14.1",
+    "sass": "1.32.10",
     "popper.js": "^1.14.3",
     "react": "^16.7.0",
     "react-copy-to-clipboard": "^5.0.1",

